Src/EB/AMReX_EB_LeastSquares_2D_K.H

```diff
@@ -498,6 +498,9 @@ Real grad_x_of_phi_on_centroids_extdir(int i,int j,int k,int n,
               continue;
             }
 
+            if (ii == phi.end.x || jj == phi.end.y)
```

Comment:
Is this to prevent out-of-bound access to phi?  Why is it only checking the end, but not the begin?

Reply:
that's a good point. i guess my test cases are not hitting that but i will update it to also check the lower bound. thanks

Reply:
Array4 has a member function for testing this
```
bool contains (int i, int j, int k) const
```
For 2d, you can pass 0 as k.